Int overflow on 32 bits arches

While building github.com/prometheus/prometheus on i686 and armv7 (w/ Golang 1.12.6):
```
BUILDSTDERR: cd /usr/share/gocode/src/github.com/cockroachdb/cockroach/pkg/util
BUILDSTDERR: /usr/lib/golang/pkg/tool/linux_arm/compile -o $WORK/b565/_pkg_.a -trimpath $WORK/b565 -shared -p github.com/cockroachdb/cockroach/pkg/util -complete -installsuffix shared -buildid x8is0fE_Soyu-Jhiysyg/x8is0fE_Soyu-Jhiysyg -goversion go1.12.6 -D "" -importcfg $WORK/b565/importcfg -pack ./every_n.go ./fast_int_map.go ./fast_int_set.go ./hash.go ./nocopy.go ./pluralize.go ./race_off.go ./reflect.go ./smalltrace.go ./strings.go ./testaddr_random.go ./topk.go ./union_find.go ./unresolved_addr.go ./unresolved_addr.pb.go
BUILDSTDERR: # github.com/cockroachdb/cockroach/pkg/util
BUILDSTDERR: /usr/share/gocode/src/github.com/cockroachdb/cockroach/pkg/util/union_find.go:92:7: constant 4294967295 overflows int
```

CC: #8179

--- a/pkg/util/union_find.go
+++ b/pkg/util/union_find.go
@@ -85,7 +85,7 @@ func (f *UnionFind) findRoot(n int) (subtreeRoot int, minElement int) {
 
 // Union joins the groups to which a and b belong.
 func (f *UnionFind) Union(a, b int) {
-	if a > math.MaxUint32 || a < 0 || b > math.MaxUint32 || b < 0 {
+	if int64(a) > math.MaxUint32 || a < 0 || int64(b) > math.MaxUint32 || b < 0 {
 		panic(fmt.Sprintf("invalid args %d, %d", a, b))
 	}
 	// Substitute a, b with the roots of the corresponding trees.

